plugins/modules/junos_bgp_global.py

```diff
@@ -1133,7 +1139,18 @@
         type: bool
   state:
     description:
-      - The state the configuration should be left in.
+    - The state the configuration should be left in.
+    - State I(purged) removes all (routing-options autonomous-system, bgp global, bgp groups, bgp family
+      and bgp neighbor family) the BGP configurations from the
+      target device. Use caution with this state.
+    - State I(deleted) only removes BGP attributes that this modules
+      manages and does not negate the BGP process completely. Thereby, preserving
+      address-family related configurations under BGP context.
+    - Running states I(deleted) and I(replaced) will result in an error if there
+      are address-family configuration lines present under a neighbor.Please use the
+      M(junipernetworks.junos.junos_bgp_af) or M(junipernetworks.junos.junos_bgp_groups)
```

Comment:
```suggestion
      M(junipernetworks.junos.junos_bgp_address_family) or M(junipernetworks.junos.junos_bgp_groups)
```

Reply:
We should also be throwing an error if the module attempts to delete a neighbor which has AF configurations like neighbor 192.168.1.10 and  neighbor 192.163.6.4 in this case:
```
ansible@an-vsrx-01# show protocols bgp | display set
set protocols bgp advertise-peer-as
set protocols bgp group external-peers neighbor 10.0.0.1 peer-as 65564
set protocols bgp group external-peers neighbor 10.0.0.2 peer-as 6556764
set protocols bgp group internal-peers type internal
set protocols bgp group internal-peers local-address 192.168.6.5
set protocols bgp group internal-peers cluster 192.168.6.5
set protocols bgp group internal-peers neighbor 192.163.6.4 family inet unicast
set protocols bgp group internal-peers neighbor 192.168.40.4
set protocols bgp group internal-peers neighbor 192.168.0.1
set protocols bgp group internal-peers neighbor 192.168.5.5
set protocols bgp group internal-peers neighbor 192.168.1.10 family inet unicast accepted-prefix-limit maximum 120
```

Reply:
this would be implemented in junos_bgp_group. not required in bgp_global.